transport/http/handler.go

```diff
@@ -117,12 +117,20 @@ func (h handler) callHandler(responseWriter *responseWriter, req *http.Request,
 		Body:            req.Body,
 		BodySize:        int(req.ContentLength),
 	}
+
+	ctx := req.Context()
+	ctx, cancel, parseTTLErr := parseTTL(ctx, treq, popHeader(req.Header, TTLMSHeader))
+	// parseTTLErr != nil is a problem only if the request is unary.
+	defer cancel()
+	ctx, span := h.createSpan(ctx, req, treq, start)
```

Comment:
The span creation has to come after header population in `treq` (L127). The span creation depends on the tracing related headers.